addons/mail/models/mail_activity.py

```diff
@@ -735,6 +736,67 @@ def _compute_activity_state(self):
             else:
                 record.activity_state = False
 
+    def _search_activity_state(self, operator, value):
+        all_states = {'overdue', 'today', 'planned', False}
+        if operator == '=':
+            states = {value}
+        elif operator == '!=':
+            states = all_states - {value}
+        elif operator == 'in':
+            states = set(value)
+        elif operator == 'not in':
+            states = all_states - set(value)
+
+        if False in states:
```

Comment:
Maybe defining your variable first helps understanding what you do, like

```
reverse_search = False in state  # make an optimized id, not in, ids domain to avoid searching on whole table
if reverse_search:
    states = all_states - state
```